### PR TITLE
Classic tbc

### PR DESCRIPTION
**Pull request by bray0u**
### Fixes
- fix unregistered Event
- fix slash command handler
- fix goto slash command
- fix zygor syntax
- fix errors with some object quest
- fix 58 boosted char starting quest lua error
- fix bugged quest mark as non complete when they are
- fix GC_Init to remember previous guide syntax

### Improvement
- add fly path using, hearthstone using
- update guidename slash command
- update GC_Init to prompt for guidename
- update /guide current guidename tooltip
- protect /guide delete guidename command

### Refactorisation
- reorganise variable declaration
- remove unused code
- set local decorator for every function

### Other
- Add "Install" section in the readme.md

_And probably some other minor stuff_

### CODE REVIEW

File: GuideCreator.lua [hunk 1]
```diff
@@ -113,24 +110,21 @@ function UpdateWindow()
...
     local questData = {}
     local questIndex = {}
     local n = GetNumQuestLogEntries()
 
     for i = 1, n do
         local title, level, suggestedGroup, isHeader, isCollapsed, isComplete, frequency, questID = GetQuestLogTitle(i)
         if questID and GetNumQuestLeaderBoards(i) > 0 then
-            questData[questID] = C_QuestLog.GetQuestObjectives(questID)
+            local qo = C_QuestLog.GetQuestObjectives(questID)
+            for key, value in pairs(qo) do
```
Comment (bray0u) [inline GuideCreator.lua:122]: This is here because of some weird quest objectives not marked as finished when they are

File: GuideCreator.lua [hunk 2]
```diff
@@ -37,51 +43,42 @@ else
...
     if not GC_GuideList then
         GC_GuideList = {}
     end
-    if UnitLevel("player") == 1 and UnitXP("player") == 0 and GetNumQuestLogEntries() == 0 then
-        GC_Settings["CurrentGuide"] = UnitName("player")
-        GC_GuideList[GC_Settings["CurrentGuide"]] = ""
-    end
-end
-
-local function pguide(...)
```
Comment (bray0u) [inline GuideCreator.lua:59]: unused

File: GuideCreator.lua [hunk 3]
```diff
@@ -113,24 +110,21 @@ function UpdateWindow()
     ScrollDown()
 end
 
-local function IsQuestComplete(quest)
```
Comment (bray0u) [inline GuideCreator.lua:116]: unused

File: GuideCreator.lua [hunk 4]
```diff
@@ -295,7 +299,10 @@ function questObjectiveComplete(id, name, obj, text, type)
     previousQuestNPC = nil
     previousQuest = id
     questEvent = "complete"
-    if not skip or (lastObj == obj and lastId == id) then
+    if lastObj ~= obj or lastId ~= id then
```
Comment (bray0u) [inline GuideCreator.lua:302]: see comment line 122

File: GuideCreator.lua [hunk 5]
```diff
@@ -546,41 +629,12 @@ eventFrame:SetScript(
                     end
                 end
             end
-
-            QuestLog = questData
-        end
-
-        if Debug == true then
```
Comment (bray0u) [inline GuideCreator.lua:553]: unsued

File: GuideCreator.lua [hunk 6]
```diff
@@ -451,84 +487,131 @@ local function FlightPath()
         local x, y = GetPlayerMapPosition("player")
         step = format("\n[G%.1f,%.1f%s]Get the [P %s] flight path", x, y, mapName, subzone)
     elseif GC_Settings["syntax"] == "Zygor" then
-        step = string.format("\nstep\n    goto %s,%.1f,%.1f\n    fpath %s", mapName, x, y, subzone)
+        step = string.format("\nstep\n  Learn new fligh tpath|.goto %s,%.1f,%.1f", subzone, x, y)
```
Comment (Zarant) [inline GuideCreator.lua:490]: this change is unnecessary and it breaks the addon syntax

**Pull request merged**